sdk/spring/spring-messaging-azure-storage-queue/src/main/java/com/azure/spring/storage/queue/core/factory/StorageQueueClientFactory.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.storage.queue.core.factory;
+
+import com.azure.storage.queue.QueueAsyncClient;
+
+/**
+ * The strategy to produce {@link QueueAsyncClient} instance.
+ */
+public interface StorageQueueClientFactory {
```

Comment:
Should we unify this class name for these factories? with event hubs and service bus?